container/join_layers.py

```diff
@@ -199,22 +256,34 @@ def main():
         diffid_to_blobsum[diff_id] = blob_sum
         blobsum_to_legacy[blob_sum] = legacy_image
 
-  for entry in args.layer:
-    elts = entry.split('=')
-    if len(elts) != 4:
-      raise Exception('Expected associative list key=value, got: %s' % entry)
-    (diffid_filename, blobsum_filename,
-     unzipped_filename, zipped_filename) = elts
-
-    diff_id = 'sha256:' + utils.ExtractValue(diffid_filename)
-    blob_sum = 'sha256:' + utils.ExtractValue(diffid_filename)
-
-    diffid_to_blobsum[diff_id] = blob_sum
-    blobsum_to_unzipped[blob_sum] = unzipped_filename
-    blobsum_to_zipped[blob_sum] = zipped_filename
+  if args.layer:
+    for entry in args.layer:
+      elts = entry.split('=')
+      if len(elts) != 4:
+        raise Exception('Expected associative list key=value, got: %s' % entry)
+      (diffid_filename, blobsum_filename,
+      unzipped_filename, zipped_filename) = elts
+
+      diff_id = 'sha256:' + utils.ExtractValue(diffid_filename)
+      blob_sum = 'sha256:' + utils.ExtractValue(blobsum_filename)
+
+      diffid_to_blobsum[diff_id] = blob_sum
+      blobsum_to_unzipped[blob_sum] = unzipped_filename
+      blobsum_to_zipped[blob_sum] = zipped_filename
+
+  # add foreign layers
```

Comment:
I've updated the source with the following:

  # add foreign layers
  #
  # Windows base images distributed by Microsoft are using foreign layers.
  # Foreign layers are not stored in the Docker repository like normal layers.
  # Instead they include a list of URLs where the layer can be downloaded.
  # This is done because Windows base images are large (2+GB).  When someone
  # pulls a Windows image, it downloads the foreign layers from those URLs
  # instead of requesting the blob from the registry.
  # When adding foreign layers through bazel, the actual layer blob is not
  # present on the system.  Instead the base image manifest is used to
  # describe the parent image layers.

Reply:
lgtm